### ISSUE TITLE
SwaggerParameterAttribute doesn't support classes

### ISSUE THREAD

**Issue by cgbeutler** [domaindrivendev/Swashbuckle.AspNetCore#1179]
We have an endpoint with many query params. We decided to move them to a class and just refer to that for the params, kinda like this:

    [HttpGet]
    public IActionResult Get([FromQuery] QueryViewModel query)
    {

Putting the SwaggerParameterAttribute on the whole class does nothing. The attribute is marked to only be usable on params:

    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
    public class SwaggerParameterAttribute : Attribute
    {

That means we have no way to use that attribute anymore to add shema comments. I'm not sure if the xml route would work here. If you know of a workaround in the mean that, that would be nice.

Fixing this is probably not trivial. I imagine the crawler code for the schema would have to be reworked.  I realize that putting query params all into a class is not very common in MVC, so this may not be a priority, unfortunately, but I'd appreciate a fix ;)

**Comment by vova-lantsov-dev** [comment]
+1 for this

We have a class:
```c#
public sealed class MyModel
{
  [FromQuery(Name = "firstName")]
  [Required, MinLength(2)]
  public string FirstName { get; set; }

  [FromQuery(Name = "lastName")]
  [Required, MinLength(2)]
  public string LastName { get; set; }
}
```
And controller's action:
```c#
[HttpGet]
public async Task<ActionResult> GetSomething(MyModel model)
{
  // here some code
}
```
But I cannot use `[SwaggerParameter]` there.
I think this issue has a major priority, because only classes (and only their properties, not fields) support built-in validation via ModelState, so it's a must-have.

**Comment by Icontech** [comment]
I have the same problem and also prefer to put my parameters into separate classes. It should be possible to fix this issue by adding a new attribute type that can be used on properties in the class that is passed to the action method. I believe a new `AnnotationParameterFilter` needs to be created which uses `PropertyInfo` instead of `ParameterInfo` when fetching the attributes from the context.

**Comment by domaindrivendev** [comment]
I'm not sure exactly when this was resolved, but I tested this functionality on `v5.6.0` and it works as desired.
